src/hotspot/share/utilities/growableArray.hpp

```diff
@@ -514,8 +515,9 @@ void GrowableArrayWithAllocator<E, Derived>::expand_to(int new_capacity) {
 
 template <typename E, typename Derived>
 void GrowableArrayWithAllocator<E, Derived>::grow(int j) {
-  // grow the array by increasing _capacity to the first power of two larger than the size we need
-  expand_to(next_power_of_2(j));
+  const size_t next_p2 = next_power_of_2((size_t)j);
+  assert(next_p2 < INT_MAX, "GrowableArray overflow (current capacity: %d)", this->_capacity);
```

Comment:
This isn't needed.  `next_power_of_2` has all the appropriate overflow checking assertions.
Just let it do it's thing with the original `int` argument.